client/client-multi/client-cpp/impl/utils/ThreadPool.cpp

```diff
@@ -16,89 +16,94 @@
 
 #include "kaa/utils/ThreadPool.hpp"
 
-#include "kaa/logging/Log.hpp"
-#include "kaa/common/exception/KaaException.hpp"
+#include <chrono>
+#include <stdexcept>
 
+#include "kaa/common/exception/KaaException.hpp"
 
 namespace kaa {
 
 class Worker {
 public:
-    Worker(ThreadPool& threadPool) : threadPool_(threadPool) {}
-    void operator() ();
+    Worker(ThreadPool& threadPool)
+        : threadPool_(threadPool)
+    { }
 
-private:
-    ThreadPool& threadPool_;
-};
-
-void Worker::operator ()()
-{
-    KAA_MUTEX_UNIQUE_DECLARE(tasksLock, threadPool_.threadPoolGuard_);
-
-    while (threadPool_.isRun_) {
-        while (threadPool_.isRun_ && threadPool_.tasks_.empty() && !threadPool_.isPendingShutdown_) {
-            KAA_CONDITION_WAIT(threadPool_.onNewTask_, tasksLock);
-        }
+    void operator() ()
```

Comment:
Just a check. You understand inlining a method inside the class declaration makes it implicitly `inline`?